### ISSUE TITLE
Use #define for addresses and constants

### ISSUE THREAD

**Issue by Animu5** [BlockWorker/EduSoC#1]
In Bootrom.S every address and constant is written into the file, this would make it unmaintainable in the future if changes were to be made.
Instead, what I would suggest is to use a header file and write something like

`#define UART_ADDRESS <whatevertheuartaddressis>`

**Comment by BlockWorker** [comment]
I'm using Venus (the same version we used in RO2) to develop and compile the assembly files - unless I'm missing something, that doesn't have any support for external files or defines. Is there a better tool for this that supports them?

**Comment by Animu5** [comment]
https://venus.cs61c.org/
This version of venus does support #define Syntax ...

At some point a classical toolchain-based flow should be established anyway, as Venus is not really that close to embedded development anyway IMHO

**Comment by BlockWorker** [comment]
That's true. Unfortunately that version of venus is incompatible with our existing programs, since it doesn't seem to support unsigned constants which we use often (like `addi xi, xj, 0xfff` doesn't work there), doesn't seem to allow hex constants for `csrrs`, and doesn't recognize `mret` at all, for example.

But I do agree that it would be good to use something better than venus in general though, if there is anything available.

**Comment by Animu5** [comment]
That is because we use a version of venus that has been extended to support these features locally, i.e., it's only running on our servers.

But yeah, I might actually go ahead and set up a flow with a Makefile .
